scripts/types/globals.d.ts

```diff
@@ -0,0 +1,43 @@
+import { User } from 'auth0'
+
+declare global {
+  /** We inject this into the top of functions, so it will be present */
+  const TEMPLATE_DATA: Record<string, any>
+  /** @tepez/auth0-rules-types declares that the global configuration object is of this type, so we extend it for all uses of configuration, including those in db scripts */
```

Comment:
This seems false though – it'll give the misleading impression that the `POSTGRES_` vars are available in rules, and that `AUTH0_CLIENT_` vars are available in db scripts. Is there a way to separate these concerns? It doesn't seem like we really need to overload the `IAuth0RuleConfiguration` global. Instead can't we just pass our own `DBConfiguration` type as a param to the db scripts?

Reply:
I could not find one. It seems like if a library declares a global, you can't overwrite that global. I could be wrong about this, but I spent some amount of time looking. I'll look at it again though, as you're right, it's pretty bad.

Reply:
Ok, so it is in fact correct that once we've imported `@typez/auth0-rules-types`, it has permanently mutated the 'global' module, and nothing I could find in much reading turned up a way to overwrite that. However, I did come up with a less misleading way to do database configuration. It is still hacky though.